### PR TITLE
PIM-4810: Error message and import status to be modified

### PR DESCRIPTION
**Pull request by ClementGautier**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | N
| Changelog updated  | Y

### CODE REVIEW

File: features/import/product/import_products_with_invalid_data.feature [hunk 1]
```diff
@@ -304,3 +304,24 @@ Feature: Execute a job
...
     And I should see "FAILED"
     And there should be 0 product
+
+  @jira https://akeneo.atlassian.net/browse/PIM-4810
+  Scenario: Correctly detach association reference when transformation fails (PIM-4810)
+    Given the following products:
+      | sku |
+      | foo |
+      | bar |
+    Given the association of type "SUBSTITUTION" between the owner product "bar" and the product "foo"
```
Comment (sumbobyboys) [inline features/import/product/import_products_with_invalid_data.feature]:
```Given the following associations for the product "SKU":
| type | product |
|  ...    | ...         |
```
Feels more natural IMHO

File: features/import/product/import_products_with_invalid_data.feature [hunk 2]
```diff
@@ -304,3 +304,24 @@ Feature: Execute a job
...
+      foo;
+      bar;foo,unknown
+      """
+    And the following job "footwear_product_import" configuration:
+      | filePath | %file to import% |
+    When I am on the "footwear_product_import" import job page
+    And I launch the import job
+    And I wait for the "footwear_product_import" job to finish
+    Then I should see "skipped 1"
+    And I should see "The \"Product\" with code \"unknown\" is unknown"
```
Comment (willy-ahva) [inline features/import/product/import_products_with_invalid_data.feature]: missing blank line

File: features/import/product/import_products_with_invalid_data.feature [hunk 3]
```diff
@@ -304,3 +304,24 @@ Feature: Execute a job
...
+    Given the following products:
+      | sku |
+      | foo |
+      | bar |
+    Given the association of type "SUBSTITUTION" between the owner product "bar" and the product "foo"
+    And the following CSV file to import:
+      """
+      sku;SUBSTITUTION-products
+      foo;
+      bar;foo,unknown
```
Comment (willy-ahva) [inline features/import/product/import_products_with_invalid_data.feature]: avoid to use foo / bar pls and use data name seems real.

**Comment by willy-ahva** [comment]
GM once comments fixed

File: features/import/product/import_products_with_invalid_data.feature (hunk 2 above)
Comment (ClementGautier) [inline features/import/product/import_products_with_invalid_data.feature]: done

File: features/import/product/import_products_with_invalid_data.feature (hunk 3 above)
Comment (jjanvier) [inline features/import/product/import_products_with_invalid_data.feature]: :+

**Comment by jjanvier** [comment]
GTM once fixed

**Pull request merged**